js/adapt-contrib-matching.js

```diff
@@ -229,83 +236,87 @@ define([
         },
 
         resetUserAnswer: function() {
-            this.model.set({_userAnswer: []});
+            this.model.set('_userAnswer', []);
         },
 
         resetQuestion: function() {
-            this.$('.matching-select option').prop('selected', false);
-            
-            this.$(".matching-item").removeClass("correct").removeClass("incorrect");
-            
+
+            this.$('.matching-item').removeClass('correct').removeClass('incorrect');
+
             this.model.set('_isAtLeastOneCorrectSelection', false);
-            
+
             var placeholder = this.model.get('placeholder');
-            
-            _.each(this.model.get("_items"), function(item, index) {
+            var resetAll = this.model.get('_shouldResetAllAnswers');
+
+            _.each(this.model.get('_items'), function(item, index) {
+                if(item._isCorrect && resetAll === false) return ;
```

Comment:
…could you get rid of that space before the semi colon?

Reply:
Isn't there also a space missing after the `if`?